Provider crashes when attempting to read a stopped service

**Describe the bug**
If a service is stopped, the provider will crash when attempting to read its state.

**To Reproduce**
Stop a service (e.g a postgres db) via dokku, then run `terraform plan` and it'll crash.

**Expected behavior**
Do not crash.

**Logs**
```
Stack trace from the terraform-provider-dokku_v0.4.2 plugin:

panic: runtime error: index out of range [1] with length 1

goroutine 59 [running]:
terraform-provider-dokku/dokku.dockerImageAndVersion(...)
        terraform-provider-dokku/dokku/util.go:72
terraform-provider-dokku/dokku.dokkuServiceRead(0xc00007d3e0, 0xc0005b8120, 0x28, 0x0)
        terraform-provider-dokku/dokku/dokku_generic_service.go:105 +0x236
terraform-provider-dokku/dokku.dokkuServiceUpdate(0xc00007d3e0, 0xc0005c1e80, 0xc0005b8120, 0x117cd8df287, 0x11d9580)
        terraform-provider-dokku/dokku/dokku_generic_service.go:248 +0x2fb
terraform-provider-dokku/dokku.dokkuPgUpdate(...)
        terraform-provider-dokku/dokku/dokku_postgres.go:49
terraform-provider-dokku/dokku.resourcePgUpdate(0xd9d068, 0xc00007d1a0, 0xc0005c1e80, 0xc932e0, 0xc0005b8120, 0xc000466cc0, 0x6e290a, 0xc0003fcb00)
        terraform-provider-dokku/dokku/resource_postgres_service.go:115 +0x6f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000182620, 0xd9cff8, 0xc0001dfdc0, 0xc0005c1e80, 0xc932e0, 0xc0005b8120, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:364 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000182620, 0xd9cff8, 0xc0001dfdc0, 0xc0005bf960, 0xc0003fcb00, 0xc932e0, 0xc0005b8120, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00011b728, 0xd9cff8, 0xc0001dfdc0, 0xc00007f2c0, 0xc0001dfdc0, 0xc734e0, 0xc0003e0800)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc00040bb20, 0xd9d0a0, 0xc0001dfdc0, 0xc0005bf880, 0xc00040bb20, 0xc0003e0870, 0xc000464ba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xc734e0, 0xc00040bb20, 0xd9d0a0, 0xc0003e0870, 0xc0003e7e00, 0x0, 0xd9d0a0, 0xc0003e0870, 0xc00043a2c0, 0x160)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000264fc0, 0xda4338, 0xc000001680, 0xc00015c600, 0xc000090c90, 0x11cbee0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000264fc0, 0xda4338, 0xc000001680, 0xc00015c600, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00003a1a0, 0xc000264fc0, 0xda4338, 0xc000001680, 0xc00015c600)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-dokku_v0.4.2 plugin crashed!
```

**Versions (please complete the following information):**
 - OS: [e.g linux]
- Terraform: [e.g v1.0.0]
- Dokku: [e.g v0.26.1]

**Additional context**
The problem is the `version` field is blank when reading the service, so we provide an empty string to `dockerImageAndVersion`, which expects there to be a colon in the provided string.

--- a/changes/unreleased/Fixed-20241225-113601.yaml
+++ b/changes/unreleased/Fixed-20241225-113601.yaml
@@ -1,0 +1,3 @@
+kind: Fixed
+body: Crash when trying to read stopped service
+time: 2024-12-25T11:36:01.076186Z

--- a/internal/provider/dokku_generic_service.go
+++ b/internal/provider/dokku_generic_service.go
@@ -98,11 +98,17 @@ func dokkuServiceRead(service *DokkuGenericService, client *goph.Client) error {
 	}
 
 	if status, ok := serviceInfo["status"]; ok {
-		service.Stopped = status == "exited"
+		service.Stopped = status == "exited" || status == "missing"
 	}
 
-	if version, ok := serviceInfo["version"]; ok {
-		service.Image, service.ImageVersion = dockerImageAndVersion(version)
+	log.Printf("[DEBUG] %s status %s", service.Name, serviceInfo["status"])
+	log.Printf("[DEBUG] %s stopped? %t", service.Name, service.Stopped)
+
+	// cannot read image & version if the service is stopped
+	if !service.Stopped {
+		if version, ok := serviceInfo["version"]; ok {
+			service.Image, service.ImageVersion = dockerImageAndVersion(version)
+		}
 	}
 
 	return nil

--- a/internal/provider/resource_postgres_service_test.go
+++ b/internal/provider/resource_postgres_service_test.go
@@ -89,6 +89,53 @@ resource "dokku_postgres_service" "test" {
 	})
 }
 
+// Previously had a problem reading stopped services
+// https://github.com/aaronstillwell/terraform-provider-dokku/issues/17
+func TestAccReadStoppedPostgresService(t *testing.T) {
+	serviceName := fmt.Sprintf("pg-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		CheckDestroy: testPgServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+resource "dokku_postgres_service" "test" {
+	name = "%s"
+}
+`, serviceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPgServiceExists("dokku_postgres_service.test"),
+				),
+			},
+			{
+				Config: fmt.Sprintf(`
+resource "dokku_postgres_service" "test" {
+	name    = "%s"
+	stopped = true
+}
+`, serviceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPgServiceExists("dokku_postgres_service.test"),
+					testAccCheckPgServiceStopped("dokku_postgres_service.test", true),
+				),
+			},
+			{
+				Config: fmt.Sprintf(`
+resource "dokku_postgres_service" "test" {
+	name    = "%s"
+	stopped = false
+}
+`, serviceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPgServiceExists("dokku_postgres_service.test"),
+					testAccCheckPgServiceStopped("dokku_postgres_service.test", false),
+				),
+			},
+		},
+	})
+}
+
 func testAccCheckPgServiceExists(n string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[n]
@@ -178,6 +225,39 @@ func testAccCheckPgServiceImageAndVersion(n string, image string, version string
 
 		if service.ImageVersion != version {
 			return fmt.Errorf("Image version expected to be %s, got %s", version, service.ImageVersion)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckPgServiceStopped(n string, isStopped bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Service ID not present")
+		}
+
+		sshClient := testAccProvider.Meta().(*goph.Client)
+
+		service := NewDokkuPostgresService(rs.Primary.ID)
+		err := dokkuPgRead(service, sshClient)
+
+		if err != nil {
+			return fmt.Errorf("Error reading pg resource %s", rs.Primary.ID)
+		}
+
+		if isStopped && !service.Stopped {
+			return fmt.Errorf("Service %s expected to be stopped, but it seems to be running", n)
+		}
+
+		if !isStopped && service.Stopped {
+			return fmt.Errorf("Service %s expected to be running, but it seems to be stopped", n)
 		}
 
 		return nil

